core/src/parse.rs

```diff
@@ -215,9 +216,33 @@ fn parse_while(input: &str) -> IResult<&str, Instr, VerboseError<&str>> {
     Ok((input, Instr::While(reg, body)))
 }
 
-fn try_each(input: &str) -> IResult<&str, Instr, VerboseError<&str>> {
+// TODO: for now throwing away spaces and comments
+// probably want too keep them when we do source mapping
+fn comment(input: &str) -> IResult<&str, &str, VerboseError<&str>> {
+    let (input, _) = preceded(
+        space0,
+        context(
+            "Comment",
+            terminated(
+                char(';'),
+                cut(tuple((take_till(|c| c == '\n'), line_ending))),
```

Comment:
Why do you need both the`c == '\n'` and `line_ending`? If possible we should just use `line_ending`, for portability (I highly doubt this thing will ever have to process a `\r`, but just in case...

Reply:
I tried to use `not_line_ending` and `many0` but was getting a bunch of weird issues. Ill just add `\r` in that check too